odoo/addons/base/models/ir_http.py

```diff
@@ -425,3 +430,7 @@ def get_web_translations_hash(self, modules, lang):
     @classmethod
     def _is_allowed_cookie(cls, cookie_type):
         return True if cookie_type == 'required' else bool(request.env.user)
+
+    @api.model
+    def _verify_request_recaptcha_token(self, action):
```

Comment:
```suggestion
    def _verify_request_recaptcha_token(self, action: str): -> bool
```
I'm usually not all for typing but we have several methods that use the `_verify` prefix, sometime they raise an error upon verification failure, sometime they return `True`/`False`. My gut is telling me it should raise an exception, which isn't the case here :D . Let's document it return a bool and not raise an exception :)
`action` in odoo most of the time mean an `ir.actions.actions` record, which isn't the case here, it's a string, taken from the `@route()` decorator (I would argue we should use the same variable name as inside `@route`). Don't bother typing it elsewhere, only in base is fine.